src/modules/core/components/TokenEditDialog/TokenEditDialog.tsx

```diff
@@ -46,28 +47,36 @@ const MSG = defineMessages({
 });
 
 interface Props {
-  addTokenFn: (address: Address) => Promise<any>;
+  updateTokens: (addresses: Address[]) => Promise<any>;
   cancel: () => void;
   close: () => void;
+  // Colony tokens
   tokens: AnyToken[];
+  // Token list from json file. Not supported on local env
+  tokensList?: AnyToken[];
+  // Colony native token addresss
+  nativeTokenAddress?: Address;
 }
 
 interface FormValues {
   tokenAddress: Address;
   description?: string;
+  tokenAddresses: Address[];
 }
 
 const validationSchema = yup.object({
-  tokenAddress: yup
-    .string()
-    .address()
-    // @todo validate against entering a duplicate address
-    .required(),
+  tokenAddress: yup.string().address(),
   tokenSymbol: yup.string().max(6),
 });
 
-const TokenEditDialog = ({ addTokenFn, tokens = [], cancel, close }: Props) => {
-  const { formatMessage } = useIntl();
+const TokenEditDialog = ({
+  updateTokens,
+  tokens = [],
+  cancel,
+  close,
+  tokensList = [],
+  nativeTokenAddress,
+}: Props) => {
   const [tokenData, setTokenData] = useState<OneToken | undefined>();
 
   const handleTokenSelect = (token: OneToken) => {
```

Comment:
The current implementation of the checkbox select in un-acceptable to be used in production.

It takes a lot of time from clicking a checkbox, until it actually gets selected:

![bug-token-list-checkbox-wait-time](https://user-images.githubusercontent.com/1193222/102320711-5c7f9f00-3f85-11eb-9f57-21f05d079990.gif)

And this problem will only get worse when we add more tokens to the list.

It seems to stem from the fact that the list re-renders every time you check a checkbox:

![bug-token-list-rerender](https://user-images.githubusercontent.com/1193222/102320890-95b80f00-3f85-11eb-8464-58d37205a1b5.gif)

Please try to improve this, because, as it stands, it cannot be merged :(

Possible avenues you can explore:
- memoizing the list
- conditionally rendering the elements
- finding another way of storing form values, other than the state (to no trigger a state update)

Reply:
This issue is still present. And not it seems to also scroll the list up when selecting a token:

![bug-tokens-list](https://user-images.githubusercontent.com/1193222/102620683-0e19fe00-4147-11eb-8e9f-e3b75e4f4b03.gif)